tests/test_fill.py

```diff
@@ -0,0 +1,78 @@
+import dpctl
+import numpy as np
+import pytest
+from dpctl.utils import ExecutionPlacementError
+from numpy.testing import assert_array_equal
+
+import dpnp as dnp
+
+
+def test_fill_non_scalar():
+    a = dnp.ones(5, dtype="i4")
+    val = dnp.ones(2, dtype="i4")
+
+    with pytest.raises(ValueError):
+        a.fill(val)
+
+    with pytest.raises(TypeError):
+        a.fill(dict())
```

Comment:
It'd be helpful to cover a string value also
```suggestion
@pytest.mark.parametrize(
    "val, error",
    [
        pytest.param(dnp.ones(2, dtype="i4"), ValueError, id="array"),
        pytest.param(dict(), TypeError, id="dictionary"),
        pytest.param("0", TypeError, id="string"),
    ]
)
def test_fill_non_scalar(val, error):
    a = dnp.ones(5, dtype="i4")

    with pytest.raises(error):
        a.fill(val)
```